Client/core/Graphics/CRenderItem.EffectCloner.cpp

```diff
@@ -31,17 +31,20 @@ CEffectCloner::CEffectCloner(CRenderItemManager* pManager)
 //
 //
 ////////////////////////////////////////////////////////////////
-CEffectWrap* CEffectCloner::CreateD3DEffect(const SString& strFile, const SString& strRootPath, bool bIsRawData, SString& strOutStatus, bool bDebug)
+CEffectWrap* CEffectCloner::CreateD3DEffect(const SString& strFile, const SString& strRootPath, bool bIsRawData, SString& strOutStatus,
+    bool bDebug, const std::vector<std::pair<SString, SString>>& macros)
```

Comment:
I thought about this `macros` thing, why dont we use a struct like 
```cpp
struct ShaderMacros 
{
   std::string name;
   std::string definition;
};
```
As its used on a lot of places.
Or, just typdef the whole thing to something like 
```cpp
typedef std::vector<std::pair<SString, SString>>& macros) ShaderMacroList_t;
```

Reply:
yes, please